File: algorithms_and_data_structures/lecture-1/find_sum_dice.py
# Задача поиска шанса выпадения определенной суммы на 3х игральных костях

def find_sum():
    result = int(input("Введите необходимую сумму: "))
    success_result = 0
    count = 0
    for i in range(1, 7):
        for j in range(1, 7):
            for k in range(1, 7):
                if (i + j + k == result):
                    success_result += 1
                count += 1
    return (success_result/count)


print(find_sum())

File: algorithms_and_data_structures/lecture-1/find_sum_dice.java
// Задача поиска шанса выпадения определенной суммы на игральный костях

public class find_sum_dice {
    public static void main(String[] args){
        System.out.println(findSum(5));
    }
    public static double findSum(int sum) {
        int count = 0;
        int successResult = 0;
        for (int i = 1; i <= 6; i++) {
            for (int j = 1; j <= 6; j++) {
                for (int k = 1; k <= 6; k++) {
                    if (i + j + k == sum) {
                        successResult++;
                    }
                    count++;
                }
            }
        }
        return ((double) successResult) / ((double) count);
    }
}
